server/middlewares/validateResourceId.js

```diff
@@ -0,0 +1,49 @@
+import models from '../models';
+
+const { Article, User, Comment } = models;
+
+const validateResourceId = (req, res, next) => {
+  const {
+    articleId,
+    userId,
+    commentId,
+  } = req.params;
+
+  const validate = async (type, id, model) => {
+    if (Number.isNaN(Number(id))) {
+    // articleId passed in is not a number
+      const nanError = new Error(`invalid id, ${type} id must be a number`);
+      nanError.status = 400;
+      return next(nanError);
+    }
+
+    let resource;
+    try {
+      resource = await model.findByPk(id);
+    } catch (err) {
+      next(err);
```

Comment:
add a return here, making this `return next(err)`. This is to simply always prevent the function from carrying on after the next statement

Reply:
Thank you @Darthrighteous. Feedback implemented.